modal/cls.py

```diff
@@ -227,6 +252,52 @@ async def _load_remote(obj: _Object, resolver: Resolver, existing_object_id: Opt
         rep = f"Ref({app_name})"
         return cls._from_loader(_load_remote, rep, is_another_app=True)
 
+    def with_options(
+        self: Type["_Cls"],
+        secrets: Collection[_Secret] = (),
+        gpu: GPU_T = None,
+        mounts: Collection[_Mount] = (),
+        volumes: Dict[Union[str, os.PathLike], _Volume] = {},
+        retries: Optional[Union[int, Retries]] = None,
+        timeout: Optional[int] = None,
+        concurrency_limit: Optional[int] = None,
+        allow_concurrent_inputs: Optional[int] = None,
+        container_idle_timeout: Optional[int] = None,
+        keep_warm: Optional[int] = None,
+        allow_background_volume_commits: bool = False,
+    ) -> "_Cls":
+        # TODO(gongy): figure out a cleaner way to clone a _Cls
+        cls = _Cls._from_loader(self, self._load, self._rep)
```

Comment:
This clone needs review

Reply:
yeah this seems a bit weird. I think you're assuming that `cls` is hydrated so you could just skip `_from_loader`.

You could probably add a simple clone method on the `_Object` base class that does `_Object.__new__(cls)` and then calls `_initialize_from_other`